code/datums/changelog/changelog.dm

```diff
@@ -14,8 +14,7 @@
 	if(.)
 		return
 	if(action == "get_month")
-		var/filename = params["filename"]
-		return ui.send_asset(file("html/changelogs/archive/" + filename))
+		return ui.send_asset(new /datum/asset/changelog_item(params["filename"]))
```

Comment:
So, i just noticed this. filename is user input, there would need to be some level of protection from spoofed values.

`sanitize_filename` would help, also append the .yml in game code and have the client code just sending the `year-month` without the `.yml`. This ensures only yml files can get sent across.